### PR TITLE
Create personal account - part 15

### PR DESCRIPTION
**Pull request by mejdoo**
## What's new in this PR?

- Refactored the logic by using one viewModel by fragment in order to ensure the single responsibility principle.

https://wearezeta.atlassian.net/browse/AN-6902

#### APK
[Download build #2151](http://10.10.124.11:8080/job/Pull%20Request%20Builder/2151/artifact/build/artifact/wire-dev-PR2844-2151.apk)

### CODE REVIEW

File: app/src/test/kotlin/com/waz/zclient/feature/auth/registration/personal/CreatePersonalAccountEmailViewModelTest.kt [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+    @Test
+    fun `given sendActivationCode is called, when the email is blacklisted then the activation code is not sent`() =
+        runBlockingTest {
+            lenient().`when`(sendEmailActivationCodeUseCase.run(any())).thenReturn(Either.Left(EmailBlacklisted))
+
+            createPersonalAccountEmailViewModel.sendActivationCode(TEST_EMAIL)
+
+            createPersonalAccountEmailViewModel.sendActivationCodeErrorLiveData.observeOnce {
+                it shouldBe EmailBlacklisted
```
Comment (gizemb) [inline app/src/test/kotlin/com/waz/zclient/feature/auth/registration/personal/CreatePersonalAccountEmailViewModelTest.kt:90]: how can we make this assertion when this liveData has the type of `ErrorMessage`?
`val sendActivationCodeErrorLiveData: LiveData<ErrorMessage> = _sendActivationCodeErrorLiveData`
whereas `EmailBlacklisted` is an instance of `Failure`?

I think we should wait for merging this PR until my unit test PRs merged, and then rebase this one, and try out again without all these "lenient"s. I'm feeling that these test cases are false-positives.

File: app/src/main/kotlin/com/waz/zclient/feature/auth/registration/personal/email/CreatePersonalAccountEmailViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+            }
+        }
+    }
+
+    private fun isNetworkConnectionFailure(failure: Failure): Boolean {
+        return if (failure is NetworkConnection) {
+            _networkConnectionErrorLiveData.value = Unit
+            true
+        } else false
+    }
```
Comment (gizemb) [inline app/src/main/kotlin/com/waz/zclient/feature/auth/registration/personal/email/CreatePersonalAccountEmailViewModel.kt:77]: seems like there's a repetition here, it'd be nice to extract this to a single class

File: app/src/main/kotlin/com/waz/zclient/feature/auth/registration/personal/email/CreatePersonalAccountEmailViewModel.kt [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        }
+    }
+
+    private fun sendActivationCodeSuccess() {
+        _sendActivationCodeSuccessLiveData.value = Unit
+    }
+
+    private fun sendActivationCodeFailure(failure: Failure) {
+        when (failure) {
+            is NetworkConnection -> _networkConnectionErrorLiveData.value = Unit
```
Comment (gizemb) [inline app/src/main/kotlin/com/waz/zclient/feature/auth/registration/personal/email/CreatePersonalAccountEmailViewModel.kt:65]: looks much better! :)

**Comment by mejdoo** [comment]
build again

**Pull request merged**